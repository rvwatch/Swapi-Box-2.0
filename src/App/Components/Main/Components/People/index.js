import React, { Component } from 'react';
import { array, func } from 'prop-types';
import Card from '../Card';

class People extends Component {
  render() {
    const createCards = this.props.people.map(person => {
      const favIds = this.props.favorites.map(fav => fav.id);
      const favStatus = favIds.includes(person.id);

      return (
        <Card
          toggleFavorites={this.props.toggleFavorites}
          cardData={person}
          isFav={favStatus}
          {...person}
          key={person.id}
        />
      );
    });
    return <section>{createCards}</section>;
  }
}

People.propTypes = {
  people: array,
  favorites: array,
  toggleFavorites: func
};

export default People;
