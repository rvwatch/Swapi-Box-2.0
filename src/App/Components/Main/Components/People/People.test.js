import React from 'react';
import { shallow } from 'enzyme';
import People from './index';
import {
  mockCleanPeople,
  mockFavorites
} from '../../../../../apiCalls/MockedData';

describe('People ', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(
      <People favorites={mockFavorites} people={mockCleanPeople} />
    );
  });
  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
