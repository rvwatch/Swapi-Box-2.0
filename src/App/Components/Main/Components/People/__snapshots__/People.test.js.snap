// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`People  renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <People
    favorites={
        Array [
            Object {
              "homeworld": "Tatooine",
              "id": 1,
              "name": "Luke Skywalker",
              "population": 200000,
              "species": "Human",
            },
            Object {
              "homeworld": "Tatooine",
              "id": 2,
              "name": "Luke Skywalker",
              "population": 200000,
              "species": "Human",
            },
            Object {
              "homeworld": "Tatooine",
              "id": 3,
              "name": "Luke Skywalker",
              "population": 200000,
              "species": "Human",
            },
          ]
    }
    people={
        Array [
            Object {
              "homeworld": undefined,
              "id": "Luke Skywalker0",
              "name": "Luke Skywalker",
              "population": undefined,
              "species": undefined,
            },
            Object {
              "homeworld": undefined,
              "id": "C-3PO1",
              "name": "C-3PO",
              "population": undefined,
              "species": undefined,
            },
            Object {
              "homeworld": undefined,
              "id": "Obi-Wan Kenobi2",
              "name": "Obi-Wan Kenobi",
              "population": undefined,
              "species": undefined,
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          people cards
</h1>,
        Array [
          <Card
            cardData={
                        Object {
                                    "homeworld": undefined,
                                    "id": "Luke Skywalker0",
                                    "name": "Luke Skywalker",
                                    "population": undefined,
                                    "species": undefined,
                                  }
            }
            homeworld={undefined}
            id="Luke Skywalker0"
            isFav={false}
            name="Luke Skywalker"
            population={undefined}
            species={undefined}
            toggleFavorites={undefined}
/>,
          <Card
            cardData={
                        Object {
                                    "homeworld": undefined,
                                    "id": "C-3PO1",
                                    "name": "C-3PO",
                                    "population": undefined,
                                    "species": undefined,
                                  }
            }
            homeworld={undefined}
            id="C-3PO1"
            isFav={false}
            name="C-3PO"
            population={undefined}
            species={undefined}
            toggleFavorites={undefined}
/>,
          <Card
            cardData={
                        Object {
                                    "homeworld": undefined,
                                    "id": "Obi-Wan Kenobi2",
                                    "name": "Obi-Wan Kenobi",
                                    "population": undefined,
                                    "species": undefined,
                                  }
            }
            homeworld={undefined}
            id="Obi-Wan Kenobi2"
            isFav={false}
            name="Obi-Wan Kenobi"
            population={undefined}
            species={undefined}
            toggleFavorites={undefined}
/>,
        ],
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "people cards",
        },
        "ref": null,
        "rendered": "people cards",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": "Luke Skywalker0",
        "nodeType": "function",
        "props": Object {
          "cardData": Object {
            "homeworld": undefined,
            "id": "Luke Skywalker0",
            "name": "Luke Skywalker",
            "population": undefined,
            "species": undefined,
          },
          "homeworld": undefined,
          "id": "Luke Skywalker0",
          "isFav": false,
          "name": "Luke Skywalker",
          "population": undefined,
          "species": undefined,
          "toggleFavorites": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "C-3PO1",
        "nodeType": "function",
        "props": Object {
          "cardData": Object {
            "homeworld": undefined,
            "id": "C-3PO1",
            "name": "C-3PO",
            "population": undefined,
            "species": undefined,
          },
          "homeworld": undefined,
          "id": "C-3PO1",
          "isFav": false,
          "name": "C-3PO",
          "population": undefined,
          "species": undefined,
          "toggleFavorites": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "Obi-Wan Kenobi2",
        "nodeType": "function",
        "props": Object {
          "cardData": Object {
            "homeworld": undefined,
            "id": "Obi-Wan Kenobi2",
            "name": "Obi-Wan Kenobi",
            "population": undefined,
            "species": undefined,
          },
          "homeworld": undefined,
          "id": "Obi-Wan Kenobi2",
          "isFav": false,
          "name": "Obi-Wan Kenobi",
          "population": undefined,
          "species": undefined,
          "toggleFavorites": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            people cards
</h1>,
          Array [
            <Card
              cardData={
                            Object {
                                          "homeworld": undefined,
                                          "id": "Luke Skywalker0",
                                          "name": "Luke Skywalker",
                                          "population": undefined,
                                          "species": undefined,
                                        }
              }
              homeworld={undefined}
              id="Luke Skywalker0"
              isFav={false}
              name="Luke Skywalker"
              population={undefined}
              species={undefined}
              toggleFavorites={undefined}
/>,
            <Card
              cardData={
                            Object {
                                          "homeworld": undefined,
                                          "id": "C-3PO1",
                                          "name": "C-3PO",
                                          "population": undefined,
                                          "species": undefined,
                                        }
              }
              homeworld={undefined}
              id="C-3PO1"
              isFav={false}
              name="C-3PO"
              population={undefined}
              species={undefined}
              toggleFavorites={undefined}
/>,
            <Card
              cardData={
                            Object {
                                          "homeworld": undefined,
                                          "id": "Obi-Wan Kenobi2",
                                          "name": "Obi-Wan Kenobi",
                                          "population": undefined,
                                          "species": undefined,
                                        }
              }
              homeworld={undefined}
              id="Obi-Wan Kenobi2"
              isFav={false}
              name="Obi-Wan Kenobi"
              population={undefined}
              species={undefined}
              toggleFavorites={undefined}
/>,
          ],
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "people cards",
          },
          "ref": null,
          "rendered": "people cards",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": "Luke Skywalker0",
          "nodeType": "function",
          "props": Object {
            "cardData": Object {
              "homeworld": undefined,
              "id": "Luke Skywalker0",
              "name": "Luke Skywalker",
              "population": undefined,
              "species": undefined,
            },
            "homeworld": undefined,
            "id": "Luke Skywalker0",
            "isFav": false,
            "name": "Luke Skywalker",
            "population": undefined,
            "species": undefined,
            "toggleFavorites": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "C-3PO1",
          "nodeType": "function",
          "props": Object {
            "cardData": Object {
              "homeworld": undefined,
              "id": "C-3PO1",
              "name": "C-3PO",
              "population": undefined,
              "species": undefined,
            },
            "homeworld": undefined,
            "id": "C-3PO1",
            "isFav": false,
            "name": "C-3PO",
            "population": undefined,
            "species": undefined,
            "toggleFavorites": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "Obi-Wan Kenobi2",
          "nodeType": "function",
          "props": Object {
            "cardData": Object {
              "homeworld": undefined,
              "id": "Obi-Wan Kenobi2",
              "name": "Obi-Wan Kenobi",
              "population": undefined,
              "species": undefined,
            },
            "homeworld": undefined,
            "id": "Obi-Wan Kenobi2",
            "isFav": false,
            "name": "Obi-Wan Kenobi",
            "population": undefined,
            "species": undefined,
            "toggleFavorites": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
